autopilot/agent_test.go

```diff
@@ -1091,3 +1091,219 @@ func TestAgentQuitWhenPendingConns(t *testing.T) {
 		t.Fatalf("unable to stop agent")
 	}
 }
+
+// respondWithScores checks that the moreChansRequest contains what we expect,
+// and responds with the given node scores.
+func respondWithScores(t *testing.T, testCtx *testContext,
+	channelBudget btcutil.Amount, existingChans, newChans int,
+	nodeScores map[NodeID]*NodeScore) {
+
+	t.Helper()
+
+	select {
+	case testCtx.constraints.moreChansResps <- moreChansResp{
+		numMore: uint32(newChans),
+		amt:     channelBudget,
+	}:
+	case <-time.After(time.Second * 3):
+		t.Fatalf("heuristic wasn't queried in time")
+	}
+
+	// The agent should query for scores using the constraints returned
+	// above. We expect the agent to use the maximum channel size when
+	// opening channels.
+	chanSize := testCtx.constraints.MaxChanSize()
+
+	select {
+	case req := <-testCtx.heuristic.nodeScoresArgs:
+		// All nodes in the graph should be potential channel
+		// candidates.
+		if len(req.nodes) != len(nodeScores) {
+			t.Fatalf("expected %v nodes, instead had %v",
+				len(nodeScores), len(req.nodes))
+		}
+
+		// One channel is already open.
+		if len(req.chans) != int(existingChans) {
+			t.Fatalf("expected 1 existing channel, got %v",
```

Comment:
Should be expected `existingChans`?

Reply:
Fixed: https://github.com/lightningnetwork/lnd/pull/2633/commits/c201211ed2772ecb6946e224177eed0812697405